pydantic/_internal/_generate_schema.py

```diff
@@ -259,6 +259,20 @@ def _model_schema(self, cls: type[BaseModel]) -> core_schema.CoreSchema:
 
             model_validators = decorators.model_validators.values()
 
+            extra_validator = None
+            if core_config.get('extra_fields_behavior', 'ignore') == 'allow':
+                for tp in (cls, *cls.__mro__):
+                    extras_annotation = cls.__annotations__.get('__pydantic_extra__', None)
+                    if extras_annotation is not None:
+                        tp = get_origin(extras_annotation)
+                        if tp not in (Dict, dict):
+                            raise PydanticSchemaGenerationError(
+                                'The type annotation for `__pydantic_extra__` must be a Dict[str, ...]'
```

Comment:
```suggestion
                                'The type annotation for `__pydantic_extra__` must be Dict[str, ...]'
```
minor, but reads better to me, feel free to ignore